File: 18_Bit_Manipultion/05_Missing_number/main.c
#include <stdio.h>

int missingNumber(int* nums, int numsSize) {
    int xor_all = 0;
    for (int i = 0; i <= numsSize; i++) {
        xor_all ^= i;
    }

    for (int i = 0; i < numsSize; i++) {
        xor_all ^= nums[i];
    }

    return xor_all;
}

int main() {
    int nums1[] = {1, 2, 3};
    int nums2[] = {0, 2};

    printf("Missing number in nums1: %d\n", missingNumber(nums1, 3)); // Output: 0
    printf("Missing number in nums2: %d\n", missingNumber(nums2, 2)); // Output: 1

    return 0;
}

File: 18_Bit_Manipultion/05_Missing_number/main.py
from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        xor_all = 0
        n = len(nums)
        
        for i in range(n + 1):
            xor_all ^= i
            
        for num in nums:
            xor_all ^= num
            
        return xor_all

# Example usage
sol = Solution()
nums1 = [1, 2, 3]
nums2 = [0, 2]

print(f"Missing number in nums1: {sol.missingNumber(nums1)}") # Output: 0
print(f"Missing number in nums2: {sol.missingNumber(nums2)}") # Output: 1
